### PR TITLE
Define data audit result structure

### PR DESCRIPTION
**Pull request by realloc**
Inner ring nodes conduct data audit and submit audit results. Results are then saved in audit smart contract for settlement routines to transfer payments.

We don't need to save the full detailed audit report, but only provide enough information for other subsystems to reward or punish storage nodes.

Closes: #109

### CODE REVIEW

File: audit/types.proto [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+option go_package = "github.com/nspcc-dev/neofs-api-go/v2/audit/grpc;audit";
+option csharp_namespace = "NeoFS.API.v2.Audit";
+
+import "refs/types.proto";
+
+// DataAuditResult keeps record of conducted Data Audits. The detailed report is
+// generated separately.
+message DataAuditResult {
+  // Epoch number when the Data Audit was conducted
+  uint64 audit_epoch = 1 [json_name = "auditEpoch"];
```
Comment (alexvanin) [inline audit/types.proto:14]: To avoid varint issues in contract I propose to use `fixed64` type for epochs.

File: audit/types.proto [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  uint32 miss = 7 [json_name = "miss"];
+
+  // Number of objects under audit stored in a way not confirming placement
+  // policy or not found at all when checking PoP
+  uint32 fail = 8 [json_name = "fail"];
+
+  // List of storage node public keys that passed at least one PDP
+  repeated bytes pass_nodes = 9 [json_name = "passNodes"];
+
+  // List of storage node public keys that passed at least one PDP
```
Comment (alexvanin) [inline audit/types.proto:44]: that **failed** at least one PDP

File: audit/types.proto [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  // containers placement policy when checking PoP
+  uint32 hit = 6 [json_name = "hit"];
+
+  // Number of objects under audit placed in suboptimal way according to the
+  // containers placement policy, but still at a satisfactory level when
+  // checking PoP
+  uint32 miss = 7 [json_name = "miss"];
+
+  // Number of objects under audit stored in a way not confirming placement
+  // policy or not found at all when checking PoP
```
Comment (alexvanin) [inline audit/types.proto:38]: Maybe we should notice that `hit`, `miss` and `fail` are counters in audit **sample** and it is not precise values? E.g. `Number of objects under audit sample stored ...`

**Review by alexvanin — CHANGES_REQUESTED** [review]

File: audit/types.proto (hunk 2 above)
Comment (realloc) [reply audit/types.proto:44]: fixed

**Review by alexvanin — APPROVED** [review]

**Review by cthulhu-rider — DISMISSED** [review]

**Review by alexvanin — APPROVED** [review]

**Pull request merged**